backends/gstreamer/src/player.rs

```diff
@@ -553,70 +553,30 @@ impl GStreamerPlayer {
                     },
                 };
 
-                // In order to make buffering/downloading work as we want, apart from
-                // setting the appropriate flags, the source needs to either:
-                //
-                // 1. be an http, mms, etc. scheme
-                // 2. report that it is "bandwidth limited".
-                //
-                // 1. is not straightforward because we are using an appsrc scheme for now.
-                // This may change in the future if we end up implementing something
-                // like a custom `servohttpsrc` or wrapping the existing appsrc
-                // in a bin src that implements http/https/data URI handlers, which
-                // is what WebKit does.
-                //
-                // For 2. we need to make appsrc handle the scheduling properties query
-                // to report that it "is bandwidth limited".
-                let src_pad = match source.get_static_pad("src") {
-                    Some(src_pad) => src_pad,
-                    None => {
-                        let _ = sender
-                            .lock()
-                            .unwrap()
-                            .send(Err(BackendError::PlayerSourceSetupFailed));
-                        return None;
-                    },
-                };
-                src_pad.add_probe(gst::PadProbeType::QUERY_UPSTREAM, |_, probe_info| {
-                    if let Some(gst::PadProbeData::Query(ref mut query)) = probe_info.data {
-                        if let QueryView::Scheduling(ref mut query) = query.view_mut() {
-                            query.set(
-                                gst::SchedulingFlags::SEQUENTIAL |
-                                gst::SchedulingFlags::BANDWIDTH_LIMITED,
-                                1, -1, 0
-                            );
-                            query.add_scheduling_modes(&[gst::PadMode::Push]);
-                            println!("QUERY {:?}", query);
-                            return gst::PadProbeReturn::Handled;
-                        }
-                    }
-                    gst::PadProbeReturn::Ok
-                });
-
                 let mut inner = inner_clone.lock().unwrap();
-                let appsrc = source
+                let servosrc = source
                     .clone()
-                    .dynamic_cast::<gst_app::AppSrc>()
-                    .expect("Source element is expected to be an appsrc!");
+                    .dynamic_cast::<ServoSrc>()
+                    .expect("Source element is expected to be a servosrc!");
 
-                appsrc.set_max_bytes(MAX_SRC_QUEUE_SIZE);
-                appsrc.set_property_block(false);
+                servosrc.set_max_bytes(MAX_SRC_QUEUE_SIZE);
+                servosrc.set_property_block(false);
 
-                appsrc.set_property_format(gst::Format::Bytes);
+                servosrc.set_property_format(gst::Format::Bytes);
```

Comment:
If these properties won't change at all, it would better to place them inside servosrc too.